core/terraform/network/dns_zones.tf

```diff
@@ -95,6 +96,23 @@ resource "azurerm_private_dns_zone_virtual_network_link" "blobcore" {
   lifecycle { ignore_changes = [tags] }
 }
 
+resource "azurerm_private_dns_zone" "dfscore" {
+  name                = "privatelink.dfs.core.windows.net"
+  resource_group_name = var.resource_group_name
+  tags                = local.tre_core_tags
+  lifecycle { ignore_changes = [tags] }
+}
+
+resource "azurerm_private_dns_zone_virtual_network_link" "dfscore" {
```

Comment:
This is usually done for situations where the core vnet needs access to resources, like for api/rp/al apps or shared services that are installed into the core network.
Since this isn't appear to be the case here, it should be removed.

Reply:
thanks for the explanation, removed.